### ISSUE TITLE
Using any web functionality causes TypeError: Cannot read properties of undefined (reading 'ok')

### ISSUE THREAD

**Issue by Leo-at-PR** [pnp/pnpjs#2821]
### What version of PnPjs library you are using

3.x

### Minor Version Number

19.0

### Target environment

All

### Additional environment details

Using NodeJS with [node-sp-auth](https://www.npmjs.com/package/node-sp-auth) for authentiation.

### Question/Request

Hi there,

This issue is following on from #2819, not sure if it's directly related. I'm having issues with carrying out any sort of action. This is the error that is thrown:
```js
file:///.../node_modules/@pnp/core/timeline.js:145
                        throw Error(`Unhandled Exception: ${args[0]}`);
                              ^

Error: Unhandled Exception: TypeError: Cannot read properties of undefined (reading 'ok')
    at Proxy.<anonymous> (.../node_modules/@pnp/core/timeline.js:145:31)
    at Proxy.error (.../node_modules/@pnp/core/timeline.js:131:23)
    at Timeout._onTimeout (.../node_modules/@pnp/queryable/queryable.js:132:22)
```
I was having trouble troubleshooting this because the program would just exit, but then I added `.using(BlobParse())` to my sp instance which is what caught the error.

I am authenticating like this:
```js
const spCreds = {
    username: process.env.SHAREPOINT_USERNAME,
    password: process.env.SHAREPOINT_PASSWORD,
};

// where SITE_URL is equivalent to "https://example.sharepoint.com/sites/My_Site"
async function initialiseSharePoint() {
    try {
        const options = await getAuth(process.env.SITE_URL, spCreds); // node-sp-auth's method
        const headers = options.headers;
        const spInstance = spfi(process.env.SITE_URL).using(InjectHeaders(headers)).using(BlobParse());

        return spInstance;
    } catch (error) {
        throw new Error("Error during authentication: " + error);
    }
}

//then initialising
const sp = await initialiseSharePoint();
```
Here, `headers` is an object with 1 k-v pair (the fedauth cookie) and looks like this:
```json
{
    "Cookie": "FedAuth=...long string..."
}
```

Whenever I try to do anything (getting a folder, info about the site, etc), the error always occurs (on the line where web is used - anything after, like the console logs, are not reached). E.g.:
```js
// where FOLDER_PATH is equivalent to "Shared Documents/Folder One/foldertwo/folderthree/folderfour"
const folderProps = await sp.web.getFolderByServerRelativePath(process.env.FOLDER_PATH).properties();
console.log("folder props: " + folderProps);

//or
// where FOLDER_ID is equivalent to ABCDEFGHIJ-1234-ABCD-1234-ABCDEFGHIJKL
const folderProps = await sp.web.getFolderById(process.env.FOLDER_ID).properties();
console.log("folder props: " + folderProps);

// or
const props = await sp.web.allProperties();
console.log("all props: " + props);
```
Pretty much: any time I try to await anything with `web`, the error occurs. A lot of this code comes from the examples from the docs, so I have a strong feeling it's something to do with the authentication; It seems like it's encouraged to use a client instead of username/pwd like I am doing. Although the authentication seems to be okay (because the cookie is returned from node-sp-auth), using it as a medium for pnp seems to not work. I made sure all the env vars have no typos. Perhaps I am using incorrect URL paths? I also tried using `encodeURI` on the env vars, but that seemed to double-encode the paths. Any ideas?

Thanks in advance.

**Comment by patrick-rodgers** [comment]
Can you share the imports you are using?

You might also spend some time in the [getting started documentation](https://pnp.github.io/pnpjs/getting-started/).

**Comment by Leo-at-PR** [comment]
The relevant imports are:
```js
import { getAuth } from "node-sp-auth";
import { InjectHeaders, BlobParse } from "@pnp/queryable";
import { spfi } from "@pnp/sp";
import "@pnp/sp/webs/index.js";
import "@pnp/sp/files/index.js";
import "@pnp/sp/folders/index.js";
import "@pnp/sp/folders/web.js";
```
I'll check out that getting started page, thanks. In the mean time, any issues with the imports? The versions I am using are as follows:
```json
"node-sp-auth": "^3.0.7",
"@pnp/sp": "3.19.0",
```

**Comment by juliemturner** [comment]
`node-sp-auth` is by a different person and not related to our library. Not saying you can't get it to work but we have examples of how to do NodeJS auth and if you're going to roll your own (which is what you're doing if you use an unrelated auth library) then you're going to have to build your own behavior to wrap it. I would suggest you follow the getting started as @patrick-rodgers  stated, and then you can specifically look at [Authenticaion in NodeJS](https://pnp.github.io/pnpjs/concepts/auth-nodejs/)

**Comment by Leo-at-PR** [comment]
@juliemturner that is fair. Every authentication example I've seen for PnPjs requires a client on Azure (or whatever it's called now). I was trying to avoid this because it feels slightly verbose for what I'm trying to do. I suppose there's no way around it? If not, I'm fine creating my own client on MS and going from there. I know I can use the MS APIs to auth this way, but that seems like it's more work than it's worth if I can just use a library like this one.

**Comment by juliemturner** [comment]
Sadly, authentication is a complicated and a required step. Where it's easier is if you're extending the platform directly, like with SharePoint Framework. In that scenario auth is significantly easier because the app registration is managed for you. You are working in NodeJS which is outside of the authentication context so you absolutely need to manage authentication and that would involve an App Registration. Maybe whatever you're trying to do doesn't need to be handled outside the platform... and then yes, you might be able to simplify things. I don't know what your scenario is and why you're building in NodeJS so I can't speak to that. 

Long story short this tool is simply a proxy for the SharePoint/Microsoft Graph REST apis that provide the framework for writing good code against those APIs.

**Comment by patrick-rodgers** [comment]
A next step might be to setup an AAD application just to do a quick test. If that works, then you know there is some issue with the auth package or the calls using the fed auth cookie.

I am fairly confident this scenario works based on our usage numbers.

**Comment by bcameron1231** [comment]
Closing this due to inactivity. If you continue to have issues please open a new issue, link to this issue, and provide any additional details available. Thanks!
